OTAnalytics/application/use_cases/track_statistics.py

```diff
@@ -40,29 +41,31 @@ def __init__(
         intersection_all_non_cutting_sections: TracksIntersectingAllNonCuttingSections,
         assigned_to_all_flows: TracksAssignedToAllFlows,
         get_all_track_ids: GetAllTrackIds,
-        inside_cutting_sections: TracksInsideCuttingSections,
-        outside_cutting_sections: TracksOnlyOutsideCuttingSections,
+        track_ids_inside_cutting_sections: TrackIdsInsideCuttingSections,
     ) -> None:
         self._intersection_all_non_cutting_sections = (
             intersection_all_non_cutting_sections
         )
         self._assigned_to_all_flows = assigned_to_all_flows
         self._get_all_track_ids = get_all_track_ids
-        self._inside_cutting_sections = inside_cutting_sections
-        self._outside_cutting_sections = outside_cutting_sections
+        self._track_ids_inside_cutting_sections = track_ids_inside_cutting_sections
 
     def get_statistics(self) -> TrackStatistics:
-        track_count_inside = len(self._inside_cutting_sections.get_ids())
-        track_count_outside = len(self._outside_cutting_sections.get_ids())
-        track_count = len(set(self._get_all_track_ids()))
+        ids_all = set(self._get_all_track_ids())
+        ids_inside_cutting_sections = self._track_ids_inside_cutting_sections()
+
+        track_count_inside = len(ids_inside_cutting_sections)
+        track_count = len(ids_all)
+
+        track_count_outside = track_count - track_count_inside
 
         track_count_inside_not_intersecting = len(
-            self._inside_cutting_sections.get_ids().difference(
+            ids_inside_cutting_sections.difference(

```

Comment:
In a typical setup (all sections inside the cutting-section) the difference between two sets would be correct. However, it is possible to define a section outside the cutting section. This might result in more tracks intersecting a section than being inside the cutting section. A share greater 100% would indicate that. @MichHeilig should we show an indicator if section are located outside the current cutting section?